### PR TITLE
Add tribal

### PR DESCRIPTION
**Pull request by lweber21**
Add tribal package to bioconda

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by lweber21** [comment]
@BiocondaBot please add label

File: recipes/tribal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - $CC -c tribal/dnapars/phylip.c -o tribal/dnapars/phylip.o -fcommon
+    - $CC -c tribal/dnapars/seq.c -o tribal/dnapars/seq.o -fcommon
+    - $CC -c tribal/dnapars/dnapars.c -o tribal/dnapars/dnapars.o -fcommon
+    - $CC tribal/dnapars/seq.o tribal/dnapars/phylip.o tribal/dnapars/dnapars.o -lm -o $PREFIX/bin/dnapars
+
+requirements:
+  build:
+    - python >=3.9,<3.11
+    - pip
+    - setuptools
```
Comment (martin-g) [inline recipes/tribal/meta.yaml:27]: python, pip and setuptools should be moved to `host`

File: recipes/tribal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - pyomo >=6.7
+    - biopython >=1.81
+
+test:
+  imports:
+    - tribal
+  commands:
+    - tribal --help
+    - tribal preprocess -h
+    - tribal fit -h
```
Comment (martin-g) [inline recipes/tribal/meta.yaml:50]: Should `dnapars` be tested too ?

File: recipes/tribal/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+about:
+  home: "https://github.com/elkebir-group/TRIBAL"
+  doc_url: "https://elkebir-group.github.io/TRIBAL"
+  license: "BSD-3-Clause"
+  summary: "TRIBAL is a package to infer B cell lineage trees from single-cell RNA sequencing data."
+
+extra:
+    channel_priority:
```
Comment (martin-g) [inline recipes/tribal/meta.yaml:59]: This is not needed.

File: recipes/tribal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - networkx >=3.1
+    - pygraphviz >=1.10
+    - ete3 >=3.1.2
+    - mafft ==7.526
+    - glpk >=5.0
+    - pyomo >=6.7
+    - biopython >=1.81
+
+test:
+  files:
```
Comment (martin-g) [inline recipes/tribal/meta.yaml:44]:
```suggestion
  source_files:
```

**Comment by lweber21** [comment]
I renamed the test script run_test.py per [conda build docs: run test script](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#run-test-script). This tests that dnapars can be run as expected via the package.  All checks have passed and I have verified `run_test.py` test completed successfully.

**Review by martin-g — APPROVED** [review]

**Pull request merged**